src/shared/components/oncoprint/DeltaUtils.spec.ts

```diff
@@ -513,4 +517,172 @@ describe("Oncoprint DeltaUtils", ()=>{
+        });
+
+        const nextSpec: IHeatmapTrackSpec = {
+            key: '',
+            label: '',
+            molecularProfileId: "profile_1",
+            molecularAlterationType: molecularAlterationType,
+            datatype: "",
+            trackGroupIndex: 1,
+            onRemove: () => {},
+            data: [
+                {profile_data: 1, study_id: "study1", uid: "uid", patient: "patient1"}, 
+                {profile_data: 2, study_id: "study1", uid: "uid", patient: "patient1"}, 
+                {profile_data: 3, study_id: "study1", uid: "uid", patient: "patient1"}
+            ]
+        };
+
+        const prevSpec = undefined;
+
+        const trackspec2trackId = () => {
+            return {
+                'TREATMENT_TRACK_1': 1,
+                'TREATMENT_TRACK_2': 2
+            };
+        };
+
+        const nextProps: IOncoprintProps = makeMinimalOncoprintProps();
+        const prevProps: IOncoprintProps = makeMinimalOncoprintProps();
+        
+        const oncoprint: OncoprintJS<any> = createStubInstance(OncoprintJS);
+
+        beforeEach(function () {
+            (oncoprint.shareRuleSet as SinonStub).resetHistory();
+        });
+        
+        it('when is new track and ruleSetId is undefined, the new trackId is set as ruleSetId', () => {
+            
+            const trackIdForRuleSetSharing = {treatment: {} as {[m:string]:number}};
+
+            (oncoprint.addTracks as SinonStub).returns([1]);
+
+            transitionHeatmapTrack(nextSpec, prevSpec, trackspec2trackId, ()=>undefined, oncoprint, nextProps, prevProps, trackIdForRuleSetSharing);
+            
+            assert.isFalse((oncoprint.shareRuleSet as SinonStub).called);
+            assert.equal(trackIdForRuleSetSharing.treatment['profile_1'], 1);
+        });
+
+        it('when is new track and ruleSetId is defined, the new trackId is set as ruleSetId', () => {
+            
+            const trackIdForRuleSetSharing = {treatment: { profile_1: 1 }};
+
+            (oncoprint.addTracks as SinonStub).returns([2]);
+
+            transitionHeatmapTrack(nextSpec, prevSpec, trackspec2trackId, ()=>undefined, oncoprint, nextProps, prevProps, trackIdForRuleSetSharing);
+            
+            assert.isFalse((oncoprint.shareRuleSet as SinonStub).called);
+            assert.equal(trackIdForRuleSetSharing.treatment['profile_1'], 2);
+        });
+
+        it('when is existing track and ruleSetId is defined, the ruleset of existing track is updated to ruleSetId', () => {
+            
+            const trackIdForRuleSetSharing = {treatment: { profile_1: 2 }};
+            const prevSpec = nextSpec;
+
+            transitionHeatmapTrack(nextSpec, prevSpec, trackspec2trackId, ()=>undefined, oncoprint, nextProps, prevProps, trackIdForRuleSetSharing);
+            
+            assert.isTrue((oncoprint.shareRuleSet as SinonStub).called);
+            assert.equal(trackIdForRuleSetSharing.treatment['profile_1'], 2);
+        });
+
+    });
```

Comment:
this testing is great!! thank you for this thoroughness

Reply:
Thnx :)